src/renderer/api/kube-object.ts

```diff
@@ -65,6 +66,33 @@ export class KubeStatus {
 
 export type IKubeMetaField = keyof IKubeObjectMetadata;
 
+/**
+ * This represents a precondition that must be satisfied before an edit is allowed
+ */
+export interface EditPrecondtion {
+  /**
+   * Checks to see if the edited version matches the original. Should return
+   * `true` if it passes the check.
+   * @param original The original version of the kube object
+   * @param edited The edited version of the kube object
+   */
+  check(original: any, edited: any): boolean;
```

Comment:
Why it's `any` now? 
`check(original: KubeObject, edited: Partial<KubeObject>)` doesn't fit for some reasons?

Reply:
I changed it to `any` because it could be moved someplace else and isn't actually tied to `KubeObject`

Reply:
Actually I have thought that is might be better to just remove them.